tests/test_gp.py

```diff
@@ -1,95 +1,259 @@
 """Test suite for GP Module"""
 
+import numpy as np
+import numpy.testing as npt
 import tensorflow as tf
-
 import pymc4 as pm
+from pymc4.gp.cov import ExpQuad
+from pymc4.gp import LatentGP, MarginalGP
+
+
+class TestLatentGP:
```

Comment:
Shifting from fixtures to class-based tests. This looks a lot more cleaner and avoids many redundant tests that were done previously...